commands/ci/debug.js

```diff
@@ -0,0 +1,99 @@
+const cli = require('heroku-cli-util')
+const Dyno = require('heroku-run').Dyno
+const co = require('co')
+const api = require('../../lib/heroku-api')
+const git = require('../../lib/git')
+const source = require('../../lib/source')
+const TestRun = require('../../lib/test-run')
+
+// Default command. Run setup, source profile.d scripts and open a bash session
+const COMMAND = 'sprettur setup && for f in .profile.d/*; do source $f; done && bash'
+
+function* run (context, heroku) {
+  const coupling = yield api.pipelineCoupling(heroku, context.app)
+  const pipeline = coupling.pipeline
+  const commit = yield git.readCommit('HEAD')
+  const sourceBlobUrl = yield cli.action('Preparing source', co(function* () {
+    return yield source.createSourceBlob(commit.ref, context, heroku)
+  }))
+
+  const pipelineRepository = yield api.pipelineRepository(heroku, pipeline.id)
```

Comment:
We should check for the pipelineRepository before creating the source blob, as the pipelineRepository fetch may fail and render the uploaded blob unused and wasteful.